lib/widgets/compose_box.dart

```diff
@@ -1304,16 +1309,17 @@ abstract class ComposeBoxState extends State<ComposeBox> {
   ComposeBoxController get controller;
 }
 
-class _ComposeBoxState extends State<ComposeBox> implements ComposeBoxState {
-  @override ComposeBoxController get controller => _controller;
-  late final ComposeBoxController _controller;
+class _ComposeBoxState extends State<ComposeBox> with PerAccountStoreAwareStateMixin<ComposeBox> implements ComposeBoxState {
+  @override ComposeBoxController get controller => _controller!;
+  ComposeBoxController? _controller;
 
   @override
-  void initState() {
-    super.initState();
+  void onNewStore() {
     switch (widget.narrow) {
       case ChannelNarrow():
-        _controller = StreamComposeBoxController();
+        final store = PerAccountStoreWidget.of(context);
+        _controller ??= StreamComposeBoxController(store: store);
```

Comment:
Does the old value of `_controller` (if non-null) need to be `dispose()`d?

Reply:
I think `_controller` shouldn't be disposed until the `ComposeBox` is disposed; we only set it once when initializing, and update its reference to the store if there is a new store.

Reply:
Ah right, I see; I read too fast and thought we were replacing the controller here, rather than updating its reference to the store.